client/ayon_nuke/plugins/publish/validate_write_nodes.py

```diff
@@ -81,6 +87,13 @@ def process(self, instance):
         if write_node is None:
             return
 
+        # gather exposed knobs to remove them from knobs check.
+        nuke_settings = instance.context.data["project_settings"]["nuke"]
+        product_type = instance.data["productType"]
+        plugin = self.product_types_mapping[product_type]
+        create_settings = nuke_settings["create"][plugin]
+        exposed_knobs = create_settings.get("exposed_knobs", [])
```

Comment:
Not sure how many knobs can be exposed - but you may want to turn this into a set if you run this _per knob_ so that the lookup is faster.